iModelCore/Bentley/BeXml/BeXml.cpp

```diff
@@ -420,15 +420,16 @@ BeXmlDomPtr BeXmlDom::CreateAndReadFromString (BeXmlStatus& xmlStatus, Utf16CP s
         characterCount = BeStringUtilities::Utf16Len(source);
 
     // this is here because the semantics of this method was changed from taking "bufferSize" to "characterCount", and I want to make sure all the old callers get an Assert.
-    // It's ok if characterCount includes the zero terminator.
+    // It's ok if characterCount includes the zero terminator although the call to xmlReadMemory must not.
 #if defined (_WIN32)
     // This uses wcslen, which is invalid on UTF-16 strings on non-Win32.
     BeAssert (charCountCloseEnough (characterCount, (WCharCP) source));
 #endif
 
-    size_t numCharsToCopy = ((0 == characterCount) || (source[characterCount - 1])) ? characterCount + 1 : characterCount; // TFS 15586
+    // Subtract if characterCount includes the null terminator.
+    size_t numCharsToCopy = (0 == characterCount || source[characterCount] != '\0') ? characterCount : characterCount - 1; // TFS 15586

```

Comment:
@ColinKerr please review this file for the XML changes caused by the new libxml requiring that the character count not include the nul terminator.